docs/_docs/customizingyourgateway.md

```diff
@@ -28,6 +28,43 @@ The protocol buffer compiler generates camelCase JSON tags that can be used with
    mux := runtime.NewServeMux(runtime.WithMarshalerOption(runtime.MIMEWildcard, &runtime.JSONPb{OrigName:false}))
    ```
 
+### Pretty-print JSON responses when queried with ?pretty
+
+You can have Elasticsearch-style `?pretty` support in your gateway's endpoints as follows:
+
+1. Wrap the ServeMux using a stdlib [`http.HandlerFunc`](https://golang.org/pkg/net/http/#HandlerFunc)
+   that translates the provided query parameter into a custom `Accept` header, and
+2. Register a pretty-printing marshaller for that MIME code.
+
+For example:
+
+```go
+mux := runtime.NewServeMux(
+	runtime.WithMarshalerOption("application/json+pretty", &runtime.JSONPb{Indent: "  "}),
+	)
+prettier := func(h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		defer h.ServeHTTP(w, r)
+		// if the "Accept" is set and NOT */* (sent by curl), keep it as-is
```

Comment:
Why does this need special casing around `*/*`?

Reply:
It doesn't, in general. However, all folks I know (small sample) would end up using `curl` on that endpoint, and `curl` sends `Accept: */*` by default. So, if we only check that an `Accept` header _was not provided_, this will not work for `curl` users.

Reply:
It doesn't, in general. However, all folks I know (small sample) would end up using `curl` on that endpoint, and `curl` sends `Accept: */*` by default. So, if we only check that an `Accept` header _was not provided_, this will not work for `curl` users.

Reply:
If figured, if you do not care (i.e. "accept `*/*`"), we might a well override your choice 😄

Reply:
What about removing this check altogether? Any `Accept` header could be overridden. when the `pretty` parameter is present, maybe? What cases would that mess with?

Reply:
Oh you're right. In this example, that's surely fine. I was thinking _better safe than sorry_; but it's fine here.